src/instructlab/model/train.py

```diff
@@ -521,20 +615,333 @@ def get_files(directory: str, pattern: str) -> list[str]:
+    # TODO: these sub-directory names are hard-coded here but they
+    # could be parameterized in config.
+    logger.debug(f"Phased Training -- Created phase directories for {path}")
+    ckpt_path = path / "checkpoints"
+    eval_cache_path = path / "eval_cache"
+
+    os.makedirs(ckpt_path)
+    os.makedirs(eval_cache_path)
+
+
+def _training_phase(
+    train_args: TrainingArgs,
+    torch_args: TorchrunArgs,
+    data_path: pathlib.Path,
+    model_override: pathlib.Path | None = None,
+    ckpt_dir: pathlib.Path | None = None,
+) -> None:
+    """A single step of end-to-end training."""
+
+    # Third Party
+    from instructlab.training import run_training  # pylint: disable=no-name-in-module
+
+    # override whatever got injected at a higher level
+    # NOTE: have to cast pathlib.Path to str because Pydantic models require this.
+    logger.debug(
+        f"Phased Training -- training phase -- Overriding data_path: {train_args.data_path} with {data_path}"
+    )
+    train_args.data_path = str(data_path)
+
+    if ckpt_dir:
+        train_args.ckpt_output_dir = str(ckpt_dir)
+
+    if model_override:
+        logger.debug(
+            f"Phased Training -- training phase -- Overriding model_path: {train_args.model_path} with {model_override}"
+        )
+        train_args.model_path = str(model_override)
+
+    logger.debug(
+        f"Phased Training -- training phase -- Current training args for phase: {pprint.pformat(train_args)}"
+    )
+
+    run_training(train_args=train_args, torch_args=torch_args)
+
+
+def _mmlu(ckpt_path: pathlib.Path) -> float:
+    # Third Party
+    from instructlab.eval.mmlu import MMLU_TASKS, MMLUEvaluator
+    import torch
+
+    evaluator = MMLUEvaluator(ckpt_path, tasks=MMLU_TASKS, few_shots=5)
+
+    mmlu_results: tuple[float, dict] = evaluator.run()
+    ckpt_score: float = mmlu_results[0]
+
+    logging.debug("Phased Training -- MMLU eval phase -- Clearing PyTorch cache")
+    torch.cuda.empty_cache()
+
+    return ckpt_score
+
+
+def _mtbench(
+    ctx,
+    ckpt_path: pathlib.Path,
+    eval_cache_path: pathlib.Path,
+    mtbench_judge: pathlib.Path,
+) -> float:
+    # TODO: run all generations in serial and then do all judgements at once to save time loading/unloading prometheus.
+
+    # dynamic import- eval code not used outside of this path.
+    # Third Party
+    from instructlab.eval.mt_bench import MTBenchEvaluator
+    import torch
+
+    # First Party
+    from instructlab.model.evaluate import launch_server
+
+    # hard-override for local insecure vLLM serving
+    ctx.params["tls_client_cert"] = None
+    ctx.params["tls_client_key"] = None
+    ctx.params["tls_client_passwd"] = None
+    ctx.params["tls_insecure"] = True
+
+    evaluator = MTBenchEvaluator(
+        model_name=str(ckpt_path),
+        judge_model_name=str(mtbench_judge),
+        output_dir=str(eval_cache_path),
+        merge_system_user_message=True,
```

Comment:
Should this be hardcoded to `True`?

Reply:
Hypothetically it should be exposed because a model *could* support system/user messages but none that we're using do. I'll make a note to open this as an issue.

Reply:
All paths that are functionally correct today do require this param to be true.  That might not be a true statement in the future.  The only reason to make this configurable today is if you want to allow train with eval to work better with lesser judge models (while not expecting accurate evals).

Reply:
I think it's fine to add NOTE/TODOs in such cases, just so we can revisit the question later when we are not in rush. I prefer merging code sprinkled with TODOs than without. `git grep TODO` later exposes the tech debt.

Reply:
++ will do